src/main/java/com/epam/cdp/m2/hw2/aggregator/Java8ParallelAggregator.java

```diff
@@ -1,23 +1,52 @@
 package com.epam.cdp.m2.hw2.aggregator;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
+import com.epam.cdp.m2.hw2.aggregator.comparator.StringLengthComparator;
 import javafx.util.Pair;
 
 public class Java8ParallelAggregator implements Aggregator {
 
     @Override
     public int sum(List<Integer> numbers) {
-        throw new UnsupportedOperationException();
+        return Optional.ofNullable(numbers)
+            .orElse(new ArrayList<>())
+            .parallelStream()
```

Comment:
as I mentioned before `parallelStream` will not be too much useful on relatively small `List`. However it might become pretty useful if the item are some long running units of work. E.g. list of some calculation or blocked operations. So every item takes its time to be processed. In this case this one could be very useful.
You probably will need own executor since default one which will be used under the hood may surprise you for some circumstances because it may be shared with others. Also always thing about ThreadLocal variables (e.g. MDC) when you are using `parallelStream`, `CompletableFuture` etc.